doc/source/schemas/alleleAnnotationmethods.rst

```diff
@@ -0,0 +1,700 @@
+  :field operationLength:
+    The number of bases that the operation runs for.
+  :type operationLength: long
+  :field referenceSequence:
+    `referenceSequence` is only used at mismatches (`SEQUENCE_MISMATCH`)
+      and deletions (`DELETE`). Filling this field replaces the MD tag.
+      If the relevant information is not available, leave this field as `null`.
+  :type referenceSequence: null|string
+
+  A structure for an instance of a CIGAR operation.
+  `FIXME: This belongs under Reads (only readAlignment refers to this)`
+
+.. avro:record:: VariantSetMetadata
+
+  :field key:
+    The top-level key.
+  :type key: string
+  :field value:
+    The value field for simple metadata.
+  :type value: string
+  :field id:
+    User-provided ID field, not enforced by this API.
+      Two or more pieces of structured metadata with identical
+      id and key fields are considered equivalent.
+      `FIXME: If it's not enforced, then why can't it be null?`
+  :type id: string
+  :field type:
+    The type of data.
+  :type type: string
+  :field number:
+    The number of values that can be included in a field described by this
+      metadata.
+  :type number: string
+  :field description:
+    A textual description of this metadata.
+  :type description: string
+  :field info:
+    Remaining structured metadata key-value pairs.
+  :type info: map<array<string>>
+
+  Optional metadata associated with a variant set.
+
+.. avro:record:: VariantSet
+
+  :field id:
+    The variant set ID.
+  :type id: string
+  :field name:
+    The variant set name.
+  :type name: null|string
+  :field datasetId:
+    The ID of the dataset this variant set belongs to.
+  :type datasetId: string
+  :field referenceSetId:
+    The ID of the reference set that describes the sequences used by the variants in this set.
+  :type referenceSetId: string
+  :field metadata:
+    Optional metadata associated with this variant set.
+      This array can be used to store information about the variant set, such as information found
+      in VCF header fields, that isn't already available in first class fields such as "name".
+  :type metadata: array<VariantSetMetadata>
+
+  A VariantSet is a collection of variants and variant calls intended to be analyzed together.
+
+.. avro:record:: CallSet
+
+  :field id:
+    The call set ID.
+  :type id: string
+  :field name:
+    The call set name.
+  :type name: null|string
+  :field sampleId:
+    The sample this call set's data was generated from.
+      Note: the current API does not have a rigorous definition of sample. Therefore, this
+      field actually contains an arbitrary string, typically corresponding to the sampleId
+      field in the read groups used to generate this call set.
+  :type sampleId: null|string
+  :field variantSetIds:
+    The IDs of the variant sets this call set has calls in.
+  :type variantSetIds: array<string>
+  :field created:
+    The date this call set was created in milliseconds from the epoch.
```

Comment:
Time formats should all be ISO8601. 
Attributes should be ```recordCreateTime``` and ```recordUpdateTime```, and the type _string_.

See https://github.com/ga4gh/schemas/blob/metadata/doc/source/metadata_intro.rst for reference/explanation.

These formats are used below (Experiment, Analysis ...).

This change should be performed throughout the schema.